### PR TITLE
Add holidays for South Africa

### PR DESCRIPTION
**Pull request by pierredup**

### CODE REVIEW

**Comment by Nielsvanpach** [comment]
Thanks! Could you rebase on main, that way I can run the tests here

File: src/Countries/SouthAfrica.php [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+            }
+        }
+
+        return array_merge($holidays, $this->variableHolidays($year));
+    }
+
+    /** @return array<string, CarbonImmutable> */
+    protected function variableHolidays(int $year): array
+    {
+        $easter = CarbonImmutable::createFromTimestamp(easter_date($year))
```
Comment (Nielsvanpach) [inline src/Countries/SouthAfrica.php:56]: Could you use `$this->easter($year)` instead?

File: tests/Countries/SouthAfricaTest.php [hunk 2]
```diff
@@ -0,0 +1,18 @@
+<?php
+
+namespace Spatie\Holidays\Tests\Countries;
+
+use Carbon\CarbonImmutable;
+use Spatie\Holidays\Holidays;
+
+it('can calculate south africa holidays', function () {
+    CarbonImmutable::setTestNowAndTimezone('2022-01-01');
```
Comment (Nielsvanpach) [inline tests/Countries/SouthAfricaTest.php:9]: Could you use 2024? That's easier to validate for correctness.

**Comment by Nielsvanpach** [comment]
There are 2 South Africa PRs, but with different results it seems. Could you both verify which one is correct? https://github.com/spatie/holidays/pull/32

**Comment by pierredup** [comment]
Both PRs looks to implement the same days. This one handles an additional edge case, where if a holiday rolls over from the Sunday to the Monday, it shouldn't be added if the Monday is already a Holiday (E.G if 25 Dec falls on a Sunday, it can't roll over to the Monday since the 26 Dec (the following Monday) will already be a holiday).

**Comment by johanmeiring** [comment]
Ah yes, the edge case that is covered here is indeed correct.  This is the more complete implementation, and should be the one that gets merged.  I'll close my PR :-)

P.S. I'll try to remember to open a PR once we know what our general election date will be, as that is very likely to be proclaimed a public holiday as well.

**Pull request merged**